new blob sounds

Retkid opened the pull request:
added new blob sounds for people to code into blob... or me to do so in a bit

ZeWaka commented:
I'd like to see actual implementation before merging.
Also, spaces in the filepath is just asking for trouble.

ZeWaka commented on code/modules/sound/soundCache.dm:
```diff
@@ -4,6 +4,41 @@
 // Note: Vox files are excluded from this to save on rsc size
 
 var/global/list/soundCache = list(
+	"sound/voice/blob/blob01.ogg"	= 'sound/voice/blob/blob01.ogg',\
+	"sound/voice/blob/blob02.ogg"	= 'sound/voice/blob/blob02.ogg',\
```
not where these belong in the hiearchy, this list is sorted

ZeWaka commented on config/admins.txt:
```diff
@@ -3,3 +3,4 @@
 #Most of the time, you'll want to make yourself a Host.
 #To do so, type your BYOND key with no spaces, capitals or other punctuation (including underscores!), a dash with spaces on both side, then the rank you want.
 #For example, the key "ZeWaka" would be given Host rank like this: zewaka - Host
+retkid - Host
```
no

ZeWaka commented on code/modules/antagonists/blob/blob_abilities.dm:
```diff
@@ -367,6 +372,7 @@
 				B3.setOvermind(owner)
 				spreadability -= R
 
+		playsound(T, "sound/voice/blob/blob[pick("01","02","03","04","05","06")].ogg", 50, 1)
```
gross syntax, use a bounded `rand()`

ZeWaka commented on code/modules/antagonists/blob/blob_abilities.dm:
```diff
@@ -248,13 +247,21 @@
 		owner.remove_ability(/datum/blob_ability/set_color)
 		owner.remove_ability(/datum/blob_ability/tutorial)
 
+//blob13 blob14=death sounds
+//blob23=deploy, building new nucleus
+//blob01-06=spread (1/6th chance each
+//blob16-18=removing blob stuff
+//succing someone into blob: 24, 25, 27 and when completed do 26
+//blob7,9 point sounds (every 10-100, cant decide which)
+//blob9 every time you gain an EVO point
```
how about.... just name them

Retkid commented:
Everything besides the sorted list and the rand was a mistake

Do I really need to use rand() [pick] is used everywhere.

ZeWaka commented:
> Do I really need to use rand() [pick] is used everywhere.

it'll lead to cleaner code, so yes

ZeWaka commented on code/modules/antagonists/blob/blob_abilities.dm:
```diff
@@ -248,14 +247,14 @@
 			var/amount = rand(20, 30)
 			src.auto_spread(startTurf, maxRange = 3, maxTurfs = amount)
-
+		owner.playsound_local(owner.loc, 'sound/voice/blob/blobdeploy1.ogg', 50, 1)
```
why blobdeploy1 if there's only 1 at all :flooshed:

Retkid replied:
Because there was a blob 2 at some point and I forgot to rename

ZeWaka commented on code/modules/antagonists/blob/blob_abilities.dm:
```diff
@@ -328,16 +328,13 @@
 					boutput(owner, "<span class='alert'>You were unable to place a bridge on [T].</span>")
-
-				return 1
+			return 1
```
i doubt this was intentional

ZeWaka replied:
still doubt.

ZeWaka replied:
x3

ZeWaka commented on code/modules/antagonists/blob/blob_abilities.dm:
```diff
@@ -328,16 +328,13 @@
 		var/obj/blob/B1 = T.can_blob_spread_here(owner, null, isadmin(owner))
 		if (!istype(B1))
 			return 1
-
 		if (!tutorial_check("spread", T))
 			return 1
-
```
y tho

Retkid replied:
Probably a mistake I thought didn't matter and thus didn't notice

ZeWaka commented on code/modules/antagonists/blob/blob_abilities.dm:
```diff
@@ -688,6 +688,7 @@
 		B.setMaterial(getInterpolatedMaterial(B.material, reinforcing.material, 0.17))
 		qdel(reinforcing)
 
+		owner.playsound_local(owner.loc, "sound/voice/blob/blob[rand(3, 4)].ogg", 50, 1)
```
why are some 3,4 and some 3,7

Retkid replied:
Because 5 6 7 aren't supposed to play

ZeWaka replied:
what does that even mean

Retkid replied:
Well you see, believe it or not, most sounds have no specific meaning. And I didn't design the sounds for each individual function. Their uses are labeled onto them retroactively.

There are some groups of sounds that I want to use outside of their maybe stated function, but with rand or pick I cant do that. Blob is just a catch all term for stuff that I used for multiple things.

Every rand I did is intentional because i want some things to play and others to not. For whatever reason I thought it needed for some to play 3 and 4 and for them to play 3-7.

ZeWaka replied:
this PR is to assign them specific meanings, no? otherwise, why not just pick a random sound for every single one of these additions

so, the solution is to name them properly, pick/rand, and not using magic number ranges

ZeWaka commented on code/modules/antagonists/blob/blob_abilities.dm:
```diff
@@ -963,6 +964,7 @@
 			owner.gen_rate_used++
 		src.deduct_bio_points()
 		src.do_cooldown()
+		owner.playsound_local(owner.loc, "sound/voice/blob/blob[rand(3, 7)].ogg", 75, 1)
```
ditto

ZeWaka commented on code/modules/antagonists/blob/blob_abilities.dm:
```diff
@@ -1119,6 +1122,13 @@
 			repeatable--
 		if (repeatable == 0)
 			owner.available_upgrades -= src
+//using this line to seperate my sounds. I think of it seperate so i try to indent it when i cant but i cannot here
+		if (prob(80))
+			owner.playsound_local(owner.loc, 'sound/voice/blob/blobup1.ogg', 50, 1)
+		else if (prob(50))
+			owner.playsound_local(owner.loc, 'sound/voice/blob/blobup2.ogg', 50, 1)
+		else
+			owner.playsound_local(owner.loc, 'sound/voice/blob/blobup3.ogg', 50, 1)
```
like we discussed, this is not how probability really works
also that comment line is weird and unnecessary

Retkid replied:
This is intentional

100 -80 = 20
50% of /0 is 10.
So else if and else both have a 20%

ZeWaka replied:
ok, comment still weird tho

ZeWaka commented on code/modules/sound/soundCache.dm:
```diff
@@ -4,7 +4,6 @@
 // Note: Vox files are excluded from this to save on rsc size
 
 var/global/list/soundCache = list(
```
you haven't added your blob sounds to the soundcache

ZeWaka commented on code/obj/blob.dm:
```diff
@@ -972,7 +979,7 @@ var/image/blob_icon_cache
 	ks_ratio = 0.5
 	sname = "slime"
-	shot_sound = 'sound/impact_sounds/Slimy_Hit_3.ogg'
+	shot_sound = 'sound/voice/blob/blobshoot5.ogg'
```
why 5

Retkid replied:
5 sounds the best and I couldn't get a proc to work here

ZeWaka replied:
the reason is because this is compile-time, if you don't want to spend the time to get it to work like your original intent was, delete the rest of the sounds and take off the 5

ZeWaka reviewed (changes requested):
why do you have a blob.zip file in voice

not so sure these belong in sound/voice/

ZeWaka commented on code/modules/antagonists/blob/blob_abilities.dm:
```diff
@@ -248,14 +247,14 @@
 
 		if (!owner.tutorial)
 			//do a little "blobsplosion"
+		//sound and sound code additions by retkid#9135, coding assitance by pali#0439 and Paul#7955
```
why comment this here?
also would better go in the PR, we track diff changes for the purposes of license-compliant attribution

ZeWaka commented on code/modules/antagonists/blob/blob_abilities.dm:
```diff
@@ -248,14 +247,14 @@
 			var/amount = rand(20, 30)
 			src.auto_spread(startTurf, maxRange = 3, maxTurfs = amount)
-
+		owner.playsound_local(owner.loc, 'sound/voice/blob/blobdeploy.ogg', 50, 1)
```
single quotes

ZeWaka commented on code/modules/antagonists/blob/blob_abilities.dm:
```diff
@@ -407,6 +408,7 @@
 		N.setMaterial(B.material)
 		B.material = null
 		qdel(B)
+		owner.playsound_local(owner.loc, 'sound/voice/blob/blobdeploy.ogg', 50, 1)
```
single quotes

ZeWaka replied:
x2

ZeWaka commented on code/modules/antagonists/blob/blob_abilities.dm:
```diff
@@ -435,7 +437,7 @@
 		if (!tutorial_check("consume", T))
 			return
-		playsound(T, 'sound/impact_sounds/Slimy_Hit_4.ogg', 50, 1)
+		owner.playsound_local(owner.loc, "sound/voice/blob/blob[rand(2, 4)].ogg", 60, 1)
```
why 2 and 4

Retkid replied:
because those are the sounds that fit it best out of the blobbuild sounds

ZeWaka commented on code/modules/antagonists/blob/blob_abilities.dm:
```diff
@@ -474,7 +476,7 @@
 			return
 
-
+		owner.playsound_local(owner.loc, "sound/voice/blob/blob[pick("deploy", "attack")].ogg", 50, 1)
```
why is attack playing deploy sounds

Retkid replied:
Because deploy was originally made to be attack but I decided to make it deploy. Since you want me to be as specific as possible stuff like this happens.

ZeWaka replied:
you can make more overarching categories that can describe both deploying and attacking for example.... blobaction or something for example

ZeWaka commented on code/modules/antagonists/blob/blob_abilities.dm:
```diff
@@ -1119,6 +1125,12 @@
 		if (repeatable == 0)
 			owner.available_upgrades -= src
+		if (prob(80))
+			owner.playsound_local(owner.loc, 'sound/voice/blob/blobup1.ogg', 50, 1)
+		else if (prob(50))
+			owner.playsound_local(owner.loc, 'sound/voice/blob/blobup2.ogg', 50, 1)
+		else
+			owner.playsound_local(owner.loc, 'sound/voice/blob/blobup3.ogg', 50, 1)
```
single quotes

Retkid replied:
There is specific meanings but these are used by several different sounds, this is the cleanest way of doing it.

ZeWaka replied:
wrong comment

ZeWaka replied:
also Big Doubt on 'cleanest way of doing it'

ZeWaka commented on code/obj/blob.dm:
```diff
@@ -993,6 +1000,7 @@ var/image/blob_icon_cache
 			if (ishuman(asshole))
 				var/mob/living/carbon/human/literal_asshole = asshole
 				literal_asshole.remove_stamina(45)
+				playsound(hit.loc, 'sound/voice/blob/blobhit.ogg', 100, 1)
```
single quotes

ZeWaka reviewed (changes requested):
why do you have unused sounds in the PR

still not so sure these belong in sound/voice/

Retkid commented:
So I removed the unused sounds and I changed all of generic blob to blobbuiild because they all refer to building though some aren't for removing/reinforcing. Which is a "overreaching category" and in my opinion makes every naming scheme perfect and self explanatory.

I removed all weird comments and added to soundcache. 

in regards to /sound/voice. I dont really care where they are as long as they are in a folder. However I think, because there is no "antag" folder and this is where antags (Wizzard and Macho Man are also in this folder) and the sounds are, in some ways implied to LITERALLY be vocalizations of the blob, it belongs here, where else would you look for the blob folder?

Retkid commented:
Swear to god I fixed that return thing 3 times.
Hopefully everything is fixed.

-
